packages/app/background-jobs-common/src/background-job-processor/processors/AbstractBackgroundJobProcessor.ts

```diff
@@ -154,8 +157,24 @@ export abstract class AbstractBackgroundJobProcessor<
   }
 
   public async start(): Promise<void> {
-    if (this.queue) return // Skip if processor is already started
+    if (this.isStarted) return // if it is already started -> skip
+    if (this.startPromise) return await this.startPromise // if it is already starting -> wait (avoid concurrent start)
```

Comment:
we can slightly reduce amount of code if we invert the condition, and do `this.startPromise = this.internalInit()` only if promise is not set, but then proceed with `await this.startPromise` either way

Reply:
Mmm, let me apply your suggestion :D